lib/cisco_node_utils/bgp.rb

```diff
@@ -36,19 +36,6 @@ def initialize(asnum, vrf='default', instantiate=true)
       create if instantiate
     end
 
-    def self.process_asnum(asnum)
-      err_msg = "BGP asnum must be either a 'String' or an" \
-                " 'Integer' object"
-      fail ArgumentError, err_msg unless asnum.is_a?(Integer) ||
-                                         asnum.is_a?(String)
-      if asnum.is_a? String
-        # Match ASDOT '1.5' or ASPLAIN '55' strings
-        fail ArgumentError unless /^(\d+|\d+\.\d+)$/.match(asnum)
```

Comment:
Sure that sounds reasonable. Of course now I get to put all those calls back in :-)